docs/arguments.md

```diff
@@ -1,5 +1,57 @@
 # Cortex Arguments Explained
 
+## Querier
+
+- `-querier.max-concurrent`
+
+   The maximum number of top-level PromQL queries that will execute at the same time, per querier process.
+
+- `-querier.query-parallelism`
+
+   This refers to database queries against the store (e.g. Bigtable or DynamoDB).  This is the max subqueries run in parallel per higher-level query.
+
+- `-querier.timeout`
+
+   The timeout for a top-level PromQL query.
+
+- `-querier.max-samples`
+
+   Maximum number of samples a single query can load into memory, to avoid blowing up on enormous queries.
+
+The next three options only apply when the querier is used together with the Query Frontend:
+
+- `-querier.frontend-address`
+
+   Address of query frontend service, used by workers to find the frontend which will give them queries to execute.
+
+- `-querier.dns-lookup-period`
+
+   How often the workers will query DNS to re-check where the frontend is.
+
+- `-querier.worker-parallelism`
+
+   Number of simultaneous queries to process, per worker process.
```

Comment:
This should be set to `max-concurrent / # query-frontend` replicas, as this is the number of goroutines which connect to each frontend in parallel. If you have `query-parallelism > (max-concurrent / # query-frontend replicas)`, then you'll have queries queueing in the queriers and not the frontend, which will affect QoS.

Reply:
I think I might express that as a constraint on `max-concurrent`